stripe/src/main/java/com/stripe/android/model/Stripe3ds2Fingerprint.kt

```diff
@@ -85,14 +86,36 @@ internal class Stripe3ds2Fingerprint private constructor(
         internal fun create(sdkData: StripeIntent.SdkData): Stripe3ds2Fingerprint {
             require(sdkData.is3ds2) { "Expected SdkData with type='stripe_3ds2_fingerprint'." }
 
-            return Stripe3ds2Fingerprint(
-                (sdkData.data[FIELD_THREE_D_SECURE_2_SOURCE] as String),
-                DirectoryServer.lookup((sdkData.data[FIELD_DIRECTORY_SERVER_NAME] as String)),
-                (sdkData.data[FIELD_SERVER_TRANSACTION_ID] as String),
-                DirectoryServerEncryption.create(
-                    (sdkData.data[FIELD_DIRECTORY_SERVER_ENCRYPTION] as Map<String, *>)
-                )
-            )
+            when (sdkData.data) {
+                is Either.Left -> {
+                    val (data: Map<String, *>) = sdkData.data
+                    return Stripe3ds2Fingerprint(
+                        (data[FIELD_THREE_D_SECURE_2_SOURCE] as String),
+                        DirectoryServer.lookup((data[FIELD_DIRECTORY_SERVER_NAME] as String)),
+                        (data[FIELD_SERVER_TRANSACTION_ID] as String),
+                        DirectoryServerEncryption.create(
+                            (data[FIELD_DIRECTORY_SERVER_ENCRYPTION] as Map<String, *>)
+                        )
```

Comment:
can we use `as?` instead of `as`

Reply:
This parsing is unchanged from how it was before. The object expects a `String` not a `String?` and it is passed to `Stripe3ds2AuthParams` which also expects a non-null `String`. We should handle this better though. Do you have a sense of what should happen if this is null?

Reply:
let's leave as is then and revisit